packages/cozy-stack-client/src/ContactsCollection.spec.js

```diff
@@ -0,0 +1,51 @@
+import CozyStackClient from './CozyStackClient'
+import ContactsCollection from './ContactsCollection'
+
+const stackMyselfResponse = {
+  data: {
+    type: 'io.cozy.contacts',
+    id: 'bf91cce0-ef48-0137-2638-543d7eb8149c',
+    attributes: {
+      fullname: 'Alice',
+      email: [{ address: 'alice@example.com', primary: true }]
+    },
+    meta: {
+      rev: '1-6516671ec'
+    }
+  }
+}
+
+describe('ContactsCollection', () => {
+  const setup = () => {
+    const stackClient = new CozyStackClient({})
+    stackClient.fetchJSON = jest
+      .fn()
+      .mockImplementation(async (method, route) => {
+        if (method === 'POST' && route === '/contacts/myself') {
+          return stackMyselfResponse
+        } else {
+          return []
+        }
+      })
+    const col = new ContactsCollection('io.cozy.contacts', stackClient)
+    return { stackClient, col }
+  }
+
+  it('should use a special route for the myself contact', async () => {
```

Comment:
Maybe we should also check that the opposite is true, eg. a normal find call uses the DocumentCollection?

Reply:
Done, in fact I had removed the test since during tests I had a V8 crash that I did not want to understand further. Turns out that I needed to mock the _index route for the test not to run in a loop. I'll try to dig further into the implementation of DocumentCollection.find to see if there is something that we could do to remove the risk of an infinite loop (ContactCollection.find -> DocumentCollection.find -> ContactCollection.find due to DocumentCollection::createIndex recalling this.find if index creation fails). I think this can be done in another PR.

Reply:
I opened https://github.com/cozy/cozy-client/issues/744 to track this issue.